app/uk/gov/hmrc/gform/lookup/LookupOptions.scala

```diff
@@ -24,6 +24,19 @@ case class LookupOptions(options: Map[LookupLabel, LookupInfo]) extends AnyVal {
 
   def keys: Iterable[LookupLabel] = options.keys
 
-  def sorted = options.toList.sortBy { case (_, LookupInfo(_, idx)) => idx }.map(_._1)
+  def sortLookupByIdx: List[(LookupLabel, LookupInfo)] =
+    options.toList.sortBy {
+      case (_, DefaultLookupInfo(_, idx))           => idx
+      case (_, CountryLookupInfo(_, idx, _, _, _))  => idx
+      case (_, CurrencyLookupInfo(_, idx, _, _, _)) => idx
+      case (_, PortLookupInfo(_, idx, _, _, _, _))  => idx
```

Comment:
Since you've defined `sealed trait LookupInfo` to have an `index`, why not to use it here?

```
case (_, option) => option.index
```